src/core/tsi/ssl_transport_security_utils.h

```diff
@@ -142,6 +142,23 @@ tsi_result SslProtectorUnprotect(const unsigned char* protected_frames_bytes,
                                  unsigned char* unprotected_bytes,
                                  size_t* unprotected_bytes_size);
 
+// Verifies the CRL signature using X509_CRL_verify
+// https://www.openssl.org/docs/man3.0/man3/X509_CRL_verify.html
+// return: 1 if valid, 0 if invalid, and -1 is there was a problem
+int verify_crl_signature(X509_CRL* crl, X509* issuer);
```

Comment:
I'm not a big fan of using an `int` as a "status" return type, since it is error prone IMO. Could we instead return `absl::Status` or a `bool`?

This puts the burden of correctly interpreting the return type of the OpenSSL APIs on us, rather than also passing this burden to the callers of this API.

Similarly below.

Reply:
My plan was to have this layer be super close to OpenSSL, so I was using their returns.
Then I was going to add another layer that used our internal object representations and returned bools or statuses - does that sound good to you?

Reply:
IIUC, we don't need to feed these values back into OpenSSL anywhere, do we?

If we don't need to be compatible with OpenSSL, then wouldn't it be safest to immediately change to a safer return type?

Please continue to push back if I missed something or if you disagree. :)

Reply:
No, but imagine we have another fn that looks like this `bool VerifyCrlSignature(OurCrlType, OurCertType);`

Then it handles the conversion of our types to call this fn, and the conversion of that output to something meaningful to grpc, and keeps the core code where this will be called cleaner by not having to do a bunch of type translations

Reply:
But wouldn't the scenario you describe be made simpler and less error-prone by having the return type of our `VerifyCrlSignature` function be `bool` instead of `int`? Said differently, is there any scenario where the caller needs to distinguish between "something was invalid" and "there was a problem"?

IMO, we should make an API like this one as safe and easy-to-use as possible. And if there is a reason to make the return type more complex in the future, we can always change it.

Reply:
I figured there might be a case where we'd want to differentiate between "signature failed verification" and "something weird happened and we couldn't verify the signature", which is what the int return allows for, and we'd convert in the middle layer.

But, I don't feel super strongly, and the code is definitely easier to reason with when we just return bools.
Since I don't have a compelling reason yet for the fine grained I'll swap to bools, if we need more control later we can switch back.

Reply:
Thank you!